dbt/adapters/teradata/connections.py

```diff
@@ -175,6 +176,25 @@ def open(cls, connection):
         # Save the transaction mode
         cls.TMODE = credentials.tmode
 
+        logger.debug("host: {}",credentials.server);
+        logger.debug("retries: {}",credentials.retries);
+
+        if credentials.retries > 0:
+            def connect():
+                connection.handle = teradatasql.connect(**kwargs)
+                connection.state = 'open'
+                return connection.handle
+
+            retryable_exceptions = [teradatasql.OperationalError,]
+
+            return cls.retry_connection(
+                connection,
+                connect=connect,
+                logger=logger,
+                retry_limit=credentials.retries,
+                retryable_exceptions=retryable_exceptions,
+            )
```

Comment:
Did you try test case like network timeout by disconnecting VPN and then reconnect VPN before number of retries expire?

Reply:
Tested this scenario. working as expected